### PR TITLE
Mark dev dependencies as such

### PR DESCRIPTION
**Pull request by jrieken**
Let's get this out first and then we can tackle rx

### CODE REVIEW

File: lib/clients/client-v1.ts [hunk 1]
```diff
@@ -17,11 +17,10 @@ export class ClientV1 extends ClientBase implements OmniSharp.Api.V1, OmniSharp.
     public observeFindimplementations: Rx.Observable<OmniSharp.Context<OmniSharp.Models.Request, OmniSharp.Models.QuickFixResponse>>;
     public observeFindsymbols: Rx.Observable<OmniSharp.Context<OmniSharp.Models.FindSymbolsRequest, OmniSharp.Models.QuickFixResponse>>;
     public observeFindusages: Rx.Observable<OmniSharp.Context<OmniSharp.Models.FindUsagesRequest, OmniSharp.Models.QuickFixResponse>>;
-    public observeGotodefinition: Rx.Observable<OmniSharp.Context<OmniSharp.Models.GotoDefinitionRequest, OmniSharp.Models.GotoDefinitionResponse>>;
+    public observeGotodefinition: Rx.Observable<OmniSharp.Context<OmniSharp.Models.Request, OmniSharp.Models.GotoDefinitionResponse>>;
```
Comment (david-driscoll) [inline lib/clients/client-v1.ts]: We're losing the changes to add the latest definitions from the server, just can't see why or how... the only commits are against the package.json

**Comment by david-driscoll** [comment]
1) No idea why the regressions happened with the last merge commit (that's strange).
2) My big question revolves around publishing / building.  I'll give it a test today.

**Comment by david-driscoll** [comment]
Tried publishing as 1.7.0 (so as not to overwrite v2), and the installed version just had the js files.  Made a few tweaks and I'm testing those now.

**Comment by david-driscoll** [comment]
First two publishes failed to have the correct package.

1.7.0 was missing the js files.
1.7.1 was missing the .ts files (or the delcarations)

Currently building 1.7.2 that should include the declarations and js files but not the .ts files (just use the compiled js source)

**Comment by david-driscoll** [comment]
https://github.com/OmniSharp/omnisharp-node-client/releases/tag/v1.7.4 seems to be a valid package, I'm just doing some validation testing.

1) Making sure types still work when developing without a symlink
2) Make sure that the client still runs when using it in atom, again without a symlink

**Comment by david-driscoll** [comment]
I've got my changes @ https://github.com/OmniSharp/omnisharp-node-client/tree/diet-publish

@jrieken let me know if you want me to push them onto `diet` and we can merge it.

**Comment by jrieken** [comment]
:+1:

**Comment by david-driscoll** [comment]
pushed the changes in, I'll see what happens with the CI tests :snail:

**Comment by david-driscoll** [comment]
Moved development only `dependencies` into `devDependencies`, initiated by #22

**Pull request merged**